Thesis.org

```diff
@@ -1,11 +1,725 @@
+*** Lens
+The previously mentioned lens in line 9 is needed to stop leaking the bookkeeping for our tail call optimization, downards into our ~Item~.
+Lenses are a popular mechanism in functional programming to, on one side, restrict the available information, while allowing changes from the restricted side to change the whole, as well.
+It does this by providing two functions: Yanker and shover. The yanker grants access to parts of the whole. If parts of that whole are changed, the shover is called to marry that part again with the whole, so the update can trickle upwards again.
+Our lens is a "first lense", because it restricts access to anything but the ~first~ element. This is relevant because when state is passed around in reacl-c, it usually comes in the form of a list where the ~first~ element is the outter and the second element is the inner state.
+The code for the ~first-lens~ can be found in listing 22.
+#+CAPTION: Elements of the list are destructured into ~first~ and ~rest~, where on the yanker side ~rest~ is discarded and on the shover side ~first~ is ignored in favor of ~small~. ~small~ represents the previously passed ~first~, which now has been updated by the restricted side.
+#+begin_src clojure
+  (defn first-lens
+    ([[first & _]]
+     first)
+    ([[_ & rest] small]
+     (vec (cons small rest))))
+#+end_src
+
+Because we don't want the internal state of our ~runner~ to leak, but we do want the state surrounding the ~runner~ to be passed down, we ignore the inner state (bookkeeping) and pass down the outter state (implicitly passed state).
+Our lense is a function with multiple arities. That means it can take different amounts of parameters. Ours being of arity of two, means it has two different signatures. A signature where it gets one parameter and a signature where it takes two parameters.
```

Comment:
The function is not of arity two but of arities 1 and 2.

Reply:
```suggestion
Our lense is a function with multiple arities. That means it can take different amounts of parameters. Ours being of arity 1 and 2, means it has two different signatures. A signature where it gets one parameter and a signature where it takes two parameters.
```
👍